packages/twenty-server/src/modules/calendar/utils/format-google-calendar-event.util.ts

```diff
@@ -1,13 +1,14 @@
-import { calendar_v3 } from 'googleapis';
+import { calendar_v3 as calendarV3 } from 'googleapis';
 import { v4 } from 'uuid';
 
 import { CalendarEventParticipantResponseStatus } from 'src/modules/calendar/standard-objects/calendar-event-participant.object-metadata';
 import { CalendarEventWithParticipants } from 'src/modules/calendar/types/calendar-event';
 
 export const formatGoogleCalendarEvent = (
-  event: calendar_v3.Schema$Event,
+  event: calendarV3.Schema$Event,
+  iCalUIDCalendarEventIdMap: Map<string, string>,
 ): CalendarEventWithParticipants => {
-  const id = v4();
+  const id = iCalUIDCalendarEventIdMap.get(event.iCalUID ?? '') ?? v4();
```

Comment:
your map shouldn't contain an "''" key either right? Not sure about this coalescing